### DISCUSSION TITLE
Setting valid_tokens from environment or secrets for the Splunk HEC source

### DISCUSSION THREAD

**Discussion by MadsRC** [vectordotdev/vector#17168] [Q&A] [answered]
I'm trying to setup a generic Splunk HEC source, where multiple sources will be able to send data. As such, I need to dynamically add and remove entries to the `valid_tokens` array in the Vector configuration (https://vector.dev/docs/reference/configuration/sources/splunk_hec/#valid_tokens).

My initial thoughts were to set that value using environment variables or a secret backend and then send Vector a `SIGHUP` whenever the tokens change.

The documentation doesn't mention sourcing an entire array from an environment variable as something one can do. The same goes for the secret backend, it seem like it just provides strings.

Is there any way to dynamically source a configuration array?

**Comment by bruceg** [comment, marked as answer]
No, there is no way currently to source a configuration array like this. It does seem like a useful feature for this source, though.

**Reply by bruceg** [reply]
Added issue #17172

**Reply by MadsRC** [reply]
Thank you @bruceg!

Sounds like I need to brush up on my Rust skills and make a PR ;)
